packages/core/src/messagingProtocols/mls/MLSService/MLSService.ts

```diff
@@ -717,17 +714,32 @@ export class MLSService extends TypedEventEmitter<Events> {
    * Commit all pending proposals for a given groupId
    *
    * @param groupId groupId of the conversation
-   * @param skipDelete if true, do not delete the pending proposals from the database
    */
-  public async commitPendingProposals({groupId, skipDelete = false}: CommitPendingProposalsParams) {
+  public async commitPendingProposals(groupId: string, shouldRetry = true): Promise<void> {
+    const groupIdBytes = Decoder.fromBase64(groupId).asBytes;
+
     try {
-      await this.commitProposals(Decoder.fromBase64(groupId).asBytes);
+      const commitBundle = await this.coreCryptoClient.commitPendingProposals(groupIdBytes);
 
-      if (!skipDelete) {
-        await this.cancelPendingProposalsTask(groupId);
+      if (commitBundle) {
+        await this.uploadCommitBundle(groupIdBytes, commitBundle);
       }
+
+      await this.cancelPendingProposalsTask(groupId);
     } catch (error) {
-      this.logger.error(`Error while committing pending proposals for groupId ${groupId}`, error);
+      if (!shouldRetry) {
+        throw error;
+      }
+
+      this.logger.warn('Failed to commit proposals, clearing the pending commit and retrying', error);
+
+      // If we failed to commit the proposals, we need to clear the pending commit and retry
+      // this is to avoid a situation where we are stuck with pending proposals that we can't commit.
+      // If there's nothing to clear the methods might throw an error, which we can ignore.
+      await this.coreCryptoClient.clearPendingCommit(groupIdBytes).catch(() => undefined);
+      await this.coreCryptoClient.clearPendingGroupFromExternalCommit(groupIdBytes).catch(() => undefined);
+
+      return this.commitPendingProposals(groupId, false);
```

Comment:
I'm not sure I understand the logic of clearing everything and retrying. Aren't we going to retry with an empty list of pending proposals (basically a no-op)?

Reply:
We're not clearing the pending proposal, but the local commit that contains the proposal (which we want to get rid of).